src/app/pages/submission-pages/data-page/data-page.component.ts

```diff
@@ -91,6 +94,11 @@ export class DataPageComponent implements OnInit {
 
   }
 
+  onDeleteFile(event, file) {
+    const fileHref = file._links.file.href;
+
+    this.fileService.deleteFile(fileHref);
```

Comment:
I think its better to do the subscription from the onDeleteFile function instead of from the services itself. Also please remove the file from files array so user see the row removed from the table once the service return success response.